services/wallet/market/market.go

```diff
@@ -63,12 +66,14 @@ func (pm *Manager) setIsConnected(value bool) {
 		if value {
 			message = "up"
 		}
-		pm.feed.Send(walletevent.Event{
-			Type:     EventMarketStatusChanged,
-			Accounts: []common.Address{},
-			Message:  message,
-			At:       time.Now().Unix(),
-		})
+		if pm.onlineChecker == nil || pm.onlineChecker.Online() {
```

Comment:
hmm I'm not really sure about this change...
1) I don't think we've got a guarantee that the Online() state will change before the providers are detected as offline/offline. This extra check will potentially inhibit the last provider down notifications and the first providers up ones.
2) The providers ARE indeed down (simply due to the fact that there's no internet connection), shouldn't the client decide what to do with that piece of info instead of not communicating the state?

Reply:
1. Agree, the solution is not robust at all. The peer count only becomes zero after a timeout (but after the connection is re-established, it becomes greater than zero quite quickly).

2. Users still see the "please check your internet connection" banner. So the chain-down notifications might be excessive. 

if 2 is OK. I see two options here:
* Run a background loop that accesses the provider's server with a delay (maybe with a fake key).
* Simply skip notifications for `net.Error`, `net.DNSError`, `net.OpError`, `tls.RecordHeaderError`, `context.DeadlineExceeded`, `http.ErrServerClosed` and "connection refused", "i/o timeout" in `toggleConnectionState`.

and I'm leaning towards the 2nd, I've done a few tests on desktop and it seems to work good

Reply:
implemented 2)

Reply:
Alternatively, we can move this logic to the client side. I can add errors (raw or as an enum) to the "down" status, so the client can decide if it wants to display the notification. I would suggest doing this as a separate task.